src/control/security/pem_test.go

```diff
@@ -0,0 +1,161 @@
+//
+// (C) Copyright 2018 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package security
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestLoadPEMData(t *testing.T) {
+
+	goodCertPath := "appdata/certs/daosCA.crt"
+	badCertPerm := "appdata/certs/badperms.crt"
+	badCertPermError := &insecureError{badCertPerm, SafeCertPerm}
+
+	testCases := []struct {
+		filename string
+		perms    os.FileMode
+		testname string
+		expected string
+	}{
+		{goodCertPath, SafeCertPerm, "GoodCert", ""},
+		{badCertPerm, SafeCertPerm, "ImproperPermissions", badCertPermError.Error()},
+	}
+
+	// Setup permissions for tests below.
+	if err := os.Chmod(goodCertPath, SafeCertPerm); err != nil {
```

Comment:
It would be better to use temporary files/directories in order to avoid polluting the repo with test files.
https://golang.org/pkg/io/ioutil/#TempDir